clients/factory.go

```diff
@@ -250,6 +257,9 @@ func (f factory) createRestClient(u string, httpClient *http.Client) (*rest.Clie
 	if f.userAgent != "" {
 		restClient.SetHeader("User-Agent", f.userAgent)
 	}
+	for headerKey, headerValue := range f.customHeaders {
```

Comment:
This might overwrite the `User-Agent` above, we should add this into the documentation on L136

Reply:
Also, due to the way the rest client (currently) works, these also could potentially overwrite any headers set on the request, such as content type.